contracts/LivepeerProtocol.sol

```diff
@@ -523,11 +539,87 @@ contract LivepeerProtocol {
+            return JobStatus.Inactive;
+        } else {
+            // A job is active if the current block is less than the job's termination block
+            return JobStatus.Active;
+        }
+    }
+
+    /*
+     * Return a job
+     * @param _jobId Job identifier
+     */
+    function getJob(uint256 _jobId) constant returns (uint256, uint256, bytes32, uint256, address, address, uint256) {
+        Job job = jobs[_jobId];
+
+        return (job.jobId, job.streamId, job.transcodingOptions, job.maxPricePerSegment, job.broadcasterAddress, job.transcoderAddress, job.endBlock);
+    }
+
+    /*
+     * End a job. Can be called by either a broadcaster or transcoder of a job
+     */
+    function endJob(uint256 _jobId) returns (bool) {
+        // Check if job exists
+        if (_jobId < 0 || _jobId >= jobs.length) throw;
+        // Check if job already has an end block
+        if (jobs[_jobId].endBlock > 0) throw;
+        // Check if called by either broadcaster or transcoder
+        if (msg.sender != jobs[_jobId].broadcasterAddress && msg.sender != jobs[_jobId].transcoderAddress) throw;
+
+        // Set set end block for job to be in 10 blocks
+        // Note: this is a placeholder value and subject to change
+        jobs[_jobId].endBlock = block.number + 10;
+    }
+
+    /*
+     * Pseudorandomly elect a currently active transcoder that charges a price per segment less than or equal to the max price per segment for a job
+     * @param _maxPricePerSegment Max price (in LPT base units) per segment of a stream
+     */
+    function electCurrentActiveTranscoder(uint256 _maxPricePerSegment) constant returns (address) {
+        // Create array to store available transcoders charging an acceptable price per segment
+        address[] memory availableTranscoders = new address[](currentActiveTranscoders.length);
+        // Keep track of the actual number of available transcoders
+        uint256 numAvailableTranscoders = 0;
+
+        for (uint256 i = 0; i < currentActiveTranscoders.length; i++) {
+            // If a transcoders charges an acceptable price per segment add it to the array of available transcoders
+            if (transcoders[currentActiveTranscoders[i].id].pricePerSegment <= _maxPricePerSegment) {
+                availableTranscoders[numAvailableTranscoders] = currentActiveTranscoders[i].id;
+                numAvailableTranscoders++;
+            }
+        }
+
+        if (numAvailableTranscoders == 0) {
+            // There is no currently available transcoder that charges a price per segment less than or equal to the max price per segment for a job
+            return address(0);
+        } else {
+            // Pseudorandomly select an available transcoder that charges an acceptable price per segment
+            return availableTranscoders[uint256(block.blockhash(block.number.sub(1))) % numAvailableTranscoders];
```

Comment:
This is fine for now. In reality I think we want to select in proportion to stake as well. I have seen techniques for generating a random # and making selection according to stake. Let's use this for now, but track as a separate issue.